pkg/ring/ring.proto

```diff
@@ -26,4 +27,5 @@ enum IngesterState {
 
 	PENDING = 2;
 	JOINING = 3;
+	PREPARING_TO_LEAVE = 4;
```

Comment:
Probably want to start adding comments to explain these states and their transitions - how is PREPARING_TO_LEAVE different from LEAVING?

Reply:
Good idea!

We send new samples to PREPARING_TO_LEAVE ingesters but not LEAVING ingesters.
(We sort out the `token.NextIngester` pointer while in PREPARING_TO_LEAVE)